dpctl/tensor/_elementwise_funcs.py

```diff
@@ -33,7 +33,7 @@
 
 Args:
     x (usm_ndarray):
-        Input array.
+        Input array, expected to have a boolean or numeric data type.
```

Comment:
technically correct language but a bit wordy. I personally left 'input array' because the type matrix is dense.

Maybe `Input array of any data type.` is acceptable? Or just leave it as `Input array`

Note that the reason behind the `expected to have [...]` language is to indicate to users what types won't have temporaries made without querying `types` from the ufunc object. This should be clarified in a document that covers the element-wise functions more in-depth.

Reply:
I noticed there is `Input array. May have any data type.` used in array API for some ufunc (like [here](https://data-apis.org/array-api/2024.12/API_specification/generated/array_api.equal.html)).
We can align and use exactly the same wording if you don't mind.

Reply:
I think that makes sense

Reply:
Updated